examples/fidelity.py

```diff
@@ -0,0 +1,82 @@
+"""Simulates the fidelity estimation.
+
+Direct Fidelity Estimation from Few Pauli Measurements
+https://arxiv.org/abs/1104.4695
+
+Practical characterization of quantum devices without tomography
+https://arxiv.org/abs/1104.3835
+"""
+
+import cirq
+import heapq
+import itertools
+import math
+import numpy
+
+
+def build_circuit():
+  qubits = cirq.LineQubit.range(3)
+  circuit = cirq.Circuit(
+      cirq.Z(qubits[0])**0.25,  # T-Gate, non Clifford.
+      cirq.X(qubits[1])**0.123,
+      cirq.X(qubits[2])**0.456)
+  return circuit, qubits
+
+
+def simulate_trace(circuit, pauli_gates, qubits):
+  simulator = cirq.DensityMatrixSimulator()
+
+  n = len(pauli_gates)
+  d = 2**n
+
+  trace = 0
+  for x in range(d):
+    xbin = numpy.binary_repr(x, width=n)
+
+    rotated_initial_state = cirq.final_wavefunction(
+        cirq.DensePauliString(pauli_gates),
+        qubit_order=qubits,
+        initial_state=x)
+
+    y = simulator.simulate(circuit, initial_state=rotated_initial_state).measurements['y']
```

Comment:
If you only need the measurement results, i.e. samples, you can use `cirq.sample` instead of asking for the full final wavefunction.

Reply:
Like this?

It seems that the function doesn't have a way to specify the initial state, so instead I pre-pended the rotations to the circuit.

Also left the old way commented out, just in case. I can remove I indeed did what you had in mind.

Leaving unresolved for now.

Reply:
Ack